src/modules/esl-share/core/esl-share-button.ts

```diff
@@ -0,0 +1,70 @@
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, boolAttr, jsonAttr, listen, prop} from '../../esl-utils/decorators';
+import {ENTER, SPACE} from '../../esl-utils/dom/keys';
+import {toAbsoluteUrl} from '../../esl-utils/misc/url';
+import {ESLShareActionRegistry} from './esl-share-action-registry';
+
+import type {ESLShareList} from './esl-share-list';
+
+function getProp<T>(name: string, targets: Record<string, any>[], fallback: T, predicate: (val: T) => boolean): T {
+  const find = targets.find((target) => predicate(target[name]));
+  return find ? find[name] : fallback;
+}
+
+export class ESLShareButton extends ESLBaseElement {
+  public static override is = 'esl-share-button';
+
+  @attr() public action: string;
+  @attr() public link: string;
+  @attr() public name: string;
+  @jsonAttr({dataAttr: true}) public additional: Record<string, any>;
+  @attr({dataAttr: true}) public shareUrl: string;
+  @attr({dataAttr: true}) public shareTitle: string;
+  @boolAttr() public unavailable: boolean;
+  @prop('transparent') public defaultBackground: string;
```

Comment:
```suggestion
  @prop('transparent') public DEFAULT_BACKGROUND: string;
  
  @attr() public action: string;
  @attr() public link: string;
  @attr() public name: string;

  @attr({dataAttr: true}) public shareUrl: string;
  @attr({dataAttr: true}) public shareTitle: string;

  @jsonAttr({dataAttr: true}) public additional: Record<string, any>;

  @boolAttr() public unavailable: boolean;
```
Super cosmetic: groping/formatting